src/test/java/io/vertx/core/TimerTest.java

```diff
@@ -446,4 +449,24 @@ public void handle(Long l) {
     });
     await();
   }
+
+  @Repeat(times = 10)
+  @Test
+  public void testRaceWhenTimerCreatedOutsideEventLoop() {
+    int numThreads = 1000;
+    int numIter = 1;
+    Thread[] threads = new Thread[numThreads];
+    AtomicInteger count = new AtomicInteger(numIter * numThreads);
+    for (int i = 0;i < numThreads;i++) {
+      Thread th = new Thread(() -> {
+        // We need something more aggressive than a millisecond for this test
+        ((VertxImpl)vertx).scheduleTimeout(((VertxImpl) vertx).getOrCreateContext(), false, 1, TimeUnit.NANOSECONDS, ignore -> {
```

Comment:
Heh! that was a nice trick to make the scheduling faster than 1ms to make the racing condition more likely :)